### ISSUE TITLE
Русские символы в консоли на английской Windows с кодовой страницей, отличающеся от CP 1251

### ISSUE THREAD

**Issue by alexkmbk** [oscript-library/opm#88]
Прикладываю скриншот с проблемой:

**Comment by nixel2007** [comment]
По идее кириллица должна нормально выводится при chcp 65001

**Comment by alexkmbk** [comment]
> По идее кириллица должна нормально выводится при chcp 65001

Да, так и есть, но у меня в конкртеном случае (в силу определенных причин) по дефолту в консоли стоит другая. Возможно моя проблема очень редкая.

**Comment by nixel2007** [comment]
Какая у вас ос?

**Comment by alexkmbk** [comment]
windows 10 pro

**Comment by alexkmbk** [comment]
https://msdn.microsoft.com/en-us/library/system.console.outputencoding(v=vs.110).aspx - можно проверять и ставить правильную кодовую страницу.

**Comment by nixel2007** [comment]
Это наверное даже больше в движок. @EvilBeaver?

**Comment by EvilBeaver** [comment]
Это скорее к микрософту, чем ко мне. Если предлагаемое решение звучит, как:

"Определить текущую кодировку консоли, и **если она не может выводить русский** то задать такую кодировку, которая сможет выводить русский", то я пас. 

И пас я из-за строчки, выделенной жирным. У меня пробелы в образовании, я не знаю, как проверить данное условие.

Хотя, с удовольствием приму пулреквест.

**Comment by EvilBeaver** [comment]
Кстати, в oscript.cfg был параметр, который регулирует выхлоп односкрипта по умолчанию. И даже ключ запуска -encoding= есть.

**Comment by alexkmbk** [comment]
@EvilBeaver  Но мы ведь точно знаем какие кодировки поддерживают кириллицу, а какие нет, я их все на память не приведу сходу, но их число конечно, а самые популярные так это должны быть 65001 и 1251

**Comment by tsukanov-as** [comment]
@alexkmbk Вопрос наверно не в тему... Но зачем вообще использовать CMD, когда в винде он заменен на повершел? Нравится страдать? )))

**Comment by EvilBeaver** [comment]
В павершеле тоже не все гладко с кодировками.

@alexkmbk проблема в том, что узнать уже установленную кодировку консоли толком нельзя из шарпа. Если приложение стартовало из консоли с кодировкой А, то приложение про это не знает. Я помню экспериментировал с этим и внятного результата не получил. Повторюсь, возможно, мне просто не повезло. Для решения задачи из первого поста рекомендую заюзать ключик -encoding или задать его глобально в конфиге oscript.cfg

**Comment by tsukanov-as** [comment]
@EvilBeaver Все там гладко. Просто использовать нужно юникод всегда

**Comment by EvilBeaver** [comment]
@tsukanov-as в условии задачи от @alexkmbk сказано что нельзя юникод и на то есть причины.

**Comment by tsukanov-as** [comment]
@EvilBeaver Ну мое дело вбросить. Страдать или нет это личный выбор каждого )

**Comment by Asakra** [comment]
Мне помогло:
![screenshot_1](https://user-images.githubusercontent.com/46216116/53850749-4e9e9b00-3ff7-11e9-879c-ab01780266e4.png)
